cibuildwheel/options.py

```diff
@@ -599,16 +601,37 @@ def globals(self) -> GlobalOptions:
         skip_config = self.reader.get("skip", env_plat=False, option_format=ListFormat(sep=" "))
         test_skip = self.reader.get("test-skip", env_plat=False, option_format=ListFormat(sep=" "))
 
+        allow_empty = args.allow_empty or strtobool(self.env.get("CIBW_ALLOW_EMPTY", "0"))
+
+        enable_groups = self.reader.get(
+            "enable", env_plat=False, option_format=ListFormat(sep=" "), env_rule=InheritRule.APPEND
+        )
+        enable = {EnableGroups(group) for group in enable_groups.split()}
+
         free_threaded_support = strtobool(
             self.reader.get("free-threaded-support", env_plat=False, ignore_empty=True)
         )
 
-        allow_empty = args.allow_empty or strtobool(self.env.get("CIBW_ALLOW_EMPTY", "0"))
-
         prerelease_pythons = args.prerelease_pythons or strtobool(
             self.env.get("CIBW_PRERELEASE_PYTHONS", "0")
         )
 
+        if free_threaded_support or prerelease_pythons:
+            msg = (
+                "free-threaded-support and prerelease-pythons should be specified by enable instead"
+            )
+            if enable:
+                raise OptionsReaderError(msg)
+            log.warning(msg)
+
+        if free_threaded_support:
+            enable.add(EnableGroups.CPythonFreeThreaded)
+        if prerelease_pythons:
+            enable.add(EnableGroups.CPythonPrerelease)
+
+        # For backwards compatibility, we are adding PyPy for now
+        enable |= {EnableGroups.PyPy}
```

Comment:
I guess this is where the question of `!pypy` came from ?
I hadn't read the PR just yet when replying.
The idea would thus be to keep PyPy enabled in 2.22 but change the default to disabled in 3.0

Reply:
Yes.

Reply:
let's start simple then without `!` then.